client/components/DropDownMenu.js

```diff
@@ -0,0 +1,49 @@
+import React, {Component} from 'react'
+import {fetchCategories} from '../store/category'
+import {connect} from 'react-redux'
+// import {Link} from 'react-router-dom'
+
+class DropDownMenu extends Component {
+  componentDidMount() {
+    this.props.fetchCategories()
+  }
+
+  handleChange(event) {
+    const value = Number(event.target.value)
+    // If the value is valid, need to redirect URL to /product/:value Route
+    if (value) console.log({value})
+  }
+
+  render() {
+    const {categories} = this.props
+
+    return (
+      <div>
+        <select name="categories" onChange={this.handleChange}>
+          <option>Select Category</option>
+          {categories ? (
+            categories.map(category => {
+              return (
+                <option key={category.id} value={category.id}>
+                  {category.title}
+                </option>
+              )
+            })
+          ) : (
+            <option>Fetching Categories</option>
```

Comment:
IMHO if the categories are loading I would just not show the select tag, I would show a spinner… but maybe this is worth researching more.

Reply:
Examples: look at CSS frameworks like materialize and bootstrap.